scripts/post-install.sh

```diff
@@ -70,35 +66,43 @@ if [[ -f /etc/redhat-release ]] || [[ -f /etc/SuSE-release ]]; then
             install_chkconfig
         fi
     fi
-elif [[ -f /etc/debian_version ]]; then
-    # Debian/Ubuntu logic
-    if [[ "$(readlink /proc/1/exe)" == */systemd ]]; then
-        install_systemd /lib/systemd/system/telegraf.service
-        deb-systemd-invoke restart telegraf.service || echo "WARNING: systemd not running."
-    else
-        # Assuming SysVinit
-        install_init
-        # Run update-rc.d or fallback to chkconfig if not available
-        if which update-rc.d &>/dev/null; then
-            install_update_rcd
+else
+    # Ownership for RHEL is set in build.py via the `rmp-attr` option.
+    # Moving these lines out of this if statement would make `rmp -V` fail after installation.
+    test -d $LOG_DIR || mkdir -p $LOG_DIR
+    chown -R -L telegraf:telegraf $LOG_DIR
+    chmod 755 $LOG_DIR
+
+    if [[ -f /etc/debian_version ]]; then
+        # Debian/Ubuntu logic
+        if [[ "$(readlink /proc/1/exe)" == */systemd ]]; then
+            install_systemd /lib/systemd/system/telegraf.service
+            deb-systemd-invoke restart telegraf.service || echo "WARNING: systemd not running."
         else
-            install_chkconfig
+            # Assuming SysVinit
+            install_init
+            # Run update-rc.d or fallback to chkconfig if not available
+            if which update-rc.d &>/dev/null; then
+                install_update_rcd
+            else
+                install_chkconfig
+            fi
+            invoke-rc.d telegraf restart
         fi
-        invoke-rc.d telegraf restart
-    fi
-elif [[ -f /etc/os-release ]]; then
-    source /etc/os-release
-    if [[ "$NAME" = "Amazon Linux" ]]; then
-        # Amazon Linux 2+ logic
-        install_systemd /usr/lib/systemd/system/telegraf.service
-    elif [[ "$NAME" = "Amazon Linux AMI" ]]; then
-        # Amazon Linux logic
-        install_init
-        # Run update-rc.d or fallback to chkconfig if not available
-        if which update-rc.d &>/dev/null; then
-            install_update_rcd
-        else
-            install_chkconfig
+    elif [[ -f /etc/os-release ]]; then
```

Comment:
Is it so for every project we maintain right? Because I am producing PRs for Telegraf and Chronograf too and I would follow the same pattern if so! Thank you :)

Reply:
It's likely this should be ported to the other projects, though the scripts have diverged over time and Telegraf tries to support more platforms.

Reply:
@danielnelson Yep, that's what I am doing right now. You can see related PRs in the link in the PR description.